pkg/services/searchV2/doc/datasource.md

```diff
@@ -0,0 +1,405 @@
+        }
+      },
+      {
+        "name": "permissions",
+        "type": "other",
+        "typeInfo": {
+          "frame": "json.RawMessage",
+          "nullable": true
+        }
+      },
+...
+    ]
+  },
+  "data": {
+    "values": [
+...
+      [
+        [
+          { "uid": "some-datasource-uid", "type": "datasource"},
+          { "uid": "PD8C576611E62080A", "type": "datasource"}
+        ]
+      ],
+      [
+        {
+          "self": {
+            "edit": true,
+            "view": true,
+            "delete": true
+          },
+          "datasource": {
+            "PD8C576611E62080A": {
+              "query": false,
+              "view": false
+            },
+            "some-datasource-uid": {
+              "query": false,
+              "view": false
+            }
+          },
+          "annotations": {
+            "PD8C576611E62080A": {
+              "view": false
+            },
+            "some-annotation-uid": {
+              "view": false
+            }
+          }
+        }
+      ],
+...
+    ]
+  }
+}
+
+```
+
+### idea #3 - alternative structure that does not assume entity id global uniqueness as a list
+
+
+```ts
+enum Action {
+    query = "query",
+    edit = "edit",
+    delete = "delete",
+    view = "view",
+} 
+
+type EntityUID = string
+
+type EntityType = string
+
+type EntityPermissions = {
+    type: "__self" | EntityType,
+    uid: string,
+    permissions: Record<Action, boolean>
+}
+
+export type Permissions = EntityPermissions[]
+```
+
+```json
+{
+  "schema": {
+    "name": "Query results",
+    "meta": {
+      ...
+    },
+    "fields": [
+      ...
+      {
+        "name": "linked_entity_uid",
+        "type": "other",
+        "typeInfo": {
+          "frame": "json.RawMessage",
+          "nullable": true
+        }
+      },
+      {
+        "name": "permissions",
+        "type": "other",
+        "typeInfo": {
+          "frame": "json.RawMessage",
+          "nullable": true
+        }
+      },
+      ...
+    ]
+  },
+  "data": {
+    "values": [
+      ...
+      [
+        [
+          {
+            "uid": "some-datasource-uid",
+            "type": "datasource"
+          },
+          {
+            "uid": "PD8C576611E62080A",
+            "type": "datasource"
+          }
+        ]
+      ],
+      [
+        [
+          {
+            "type": "__self",
+            "uid": "abc123",
+            "permissions": {
+              "edit": true,
+              "view": true,
+              "delete": true
+            }
+          },
+          {
+            "type": "annotation",
+            "uid": "abc123",
+            "permissions": {
+              "edit": true,
+              "view": true,
+              "delete": true
+            }
+          },
+          {
+            "type": "datasource",
+            "uid": "abc123",
+            "permissions": {
+              "view": true,
+              "query": true
+            }
+          }
+        ]
```

Comment:
I added an alternative structure for the permission JSON and I think I like it more - lets discuss today @grafana/grafana-edge-squad